### DISCUSSION TITLE
Change speed stepper motor

### DISCUSSION THREAD

**Discussion by Jjjigu** [kachurovskiy/nanoels#103] [General]
Hello, congratulations on your work, els is great. 
How can I change the scroll speed when the left or right buttons are pressed? Thank you.

**Comment by Jjjigu** [comment]
I want to lower the stepper motor speed because I think that in my case it is too high, I tell you: I have a nema 34 mounted with 0.75 reduction, 400 steps in the driver and it is configured at 300 in els. the problem comes when making threads: when I move the carriage to the right stop, the sharp blow that the motor gives when stopping makes the thread start a little more to the right not making the thread correctly. If I make the threads reversing the direction of rotation of the spindle, the stop is smooth and it does it perfectly, therefore understanding that the problem may be in the speed of the motor when moving to the right

**Reply by kachurovskiy** [reply]
Hi @Jjjigu, the problem with sharp stops indeed exists and I hope to make them smooth eventually but it might not be soon.

For your particular case, I would recommend having starting limit (R when cutting a classic thread) set a bit further to the right to give the leadscrew some distance to take out the backlash. I realize that this is not always possible due to e.g. rear support being in the way.

If you want to slow down the button move left/right speed, increase `PULSE_MIN_US` - https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L16

By default it's `round(500.0 * 200.0 / MOTOR_STEPS)`, try making it `round(750.0 * 200.0 / MOTOR_STEPS)`.

If none of this helps, please post a video showcasing your problem - ideally showing which buttons you press, how carriage moves and what is cut. Thanks!

**Comment by Jjjigu** [comment]
Hello, thank you very much for the response. I will try both options that you tell me. 
This weekend I will try to record a video although I find it very difficult to appreciate in a video and it must be tenths. 
As alternative data when threading with only the stop L when it stops at that stop and I move the leadscrew with the right button to the position that seems good to me and I release that button a very abrupt change of direction occurs which makes the stop l each time make it a little more to the right and therefore making a thread useless.
Thank

**Comment by Jjjigu** [comment]
Hello, 
I have been checking the lathe in search of mechanical problems. I have found backlash in the leadscrew, more specifically in the bushings, and I am trying to repair it since the problem most likely comes from there, since when I was stopping at the Right side I understand that the leadscrew moved to the right. 
As soon as I have not repaired, I will do the tests and I will tell you, however, lowering the speed, it seems that it did not do it. I imagine that because the right stop stroke was less.
Thank you.

**Reply by kachurovskiy** [reply]
Backlash should not be a problem for lathe operation, especially if it's consistent across the length of the leadscrew. It would really help if you could post a photo or a video of the problem, it might be something simple. Cheers!

**Comment by Jjjigu** [comment]
Sorry my english is bad. Not backlash is loosness (holgura en español) in leadscrew .
I have the lathe disassembled solving that bushing.

**Comment by Jjjigu** [comment]
Hello, I think the problem is already solved.
 As I told you, there was slack in the leadscrew. I have put some bushings on it. I have tested it and apparently it works well. I will disassemble it again and put some axial bearings to solve the problem completely, but the first test has been satisfactory. 
ELS works fine. 
I have also lowered the movement speed a bit since in my case it was going very fast due to the reductions.
I am very happy with els. Congratulations.
Thank you

**Comment by Jjjigu** [comment]
![IMG_20230130_063330_1 (1)](https://user-images.githubusercontent.com/121951043/215401938-a68be584-290c-4e11-8655-e61ca30b13eb.jpg)

Hello, I have a new problem maybe related to the previous one.
Encoder have reduction 1-2 , is set to 300.
Stepper is set to 300 with 400 microsteps and reducion 0.75
The threads of the thread do not keep the same distance, I attach photos.
Thanks
![IMG_20230130_055943 (1)](https://user-images.githubusercontent.com/121951043/215400863-6c76602b-28d8-464f-a27c-770a70105868.jpg)
![IMG_20230130_065903](https://user-images.githubusercontent.com/121951043/215400917-e242e83e-0dc0-4074-a31f-565c513938ec.jpg)
![IMG_20230130_065846_1 (1)](https://user-images.githubusercontent.com/121951043/215401055-4905e286-c77d-43bd-a02a-472795f86c81.jpg)

**Reply by kachurovskiy** [reply]
Hi @Jjjigu, I've seen something like this on my closed-loop stepper when running it close to it's speed limit with factory current/gain configuration.

1. Please post a picture showing how your stepper driver is configured. Perhaps we could adjust something there.
2. Please let us know how much RPM were you running the spindle
3. Try switching the stepper into the open-loop mode. Never happened on my open-loop stepper.
4. Try lowering the spindle RPM

![image](https://user-images.githubusercontent.com/517919/215432485-bcd57a51-d0b8-454a-abbc-9def26b31442.png)

Cheers!

**Comment by Jjjigu** [comment]
Hi, thanks for answer.
The rpm 200-250 in thread 1.25, i have tested in 100 rpm, and i have tested with the hand move the spindle and result the same.
The stepper driver all pins ON(attach picture) 2.4A peak 2.0A RMS, 400 steps and full current, i think that is open-loop.
I have test with 800 steps and the result is the same.
Attach picture of nema 34, conected to 60v, 8-wire unipolar with four coils connected in parallel.
Thank
![Screenshot_2023-01-30-10-57-04-762_com wallapop](https://user-images.githubusercontent.com/121951043/215446644-3e0cf46e-3f13-4562-805a-60641f7cc580.jpg)
![IMG_20230130_110705](https://user-images.githubusercontent.com/121951043/215447687-5b4ff677-6be7-4dd9-9b87-748b6668f498.jpg)
![IMG_20230130_072635](https://user-images.githubusercontent.com/121951043/215449264-7b6bc21c-a7db-4469-970b-5ec45e4a5ee7.jpg)

**Reply by kachurovskiy** [reply]
Got it. Perhaps we should first check if the carriage moves are correct before going into threading. Can you please put an indicator on the side of the carriage, use the left/right buttons and see if with each step it moves the expected amount? Thank you

**Comment by Jjjigu** [comment]
Hi, attach 2 video els ON and OFF, but i cant upload. How can i send ?
Thanks

**Reply by kachurovskiy** [reply]
If you can't upload them here (drag and drop in the comments) them uploading them to e.g. Google Drive and sharing a link would also work.

**Comment by Jjjigu** [comment]
Thanks, 
you can see that ELS OFF i move 1mm and carriage move first 0.9mm second 1.1mm third 0.9mm -  1.1mm   -  0.9mm 
https://drive.google.com/file/d/19x-CRtyObvWjxBfO8Zpq5WHVxzAgrdjV/view?usp=share_link
https://drive.google.com/file/d/1A2zZOEUjfCmFTNZkNjeSEDY0B8tKFFsF/view?usp=share_link

**Reply by kachurovskiy** [reply]
You're moving in the ON mode so it tries to stay in the thread and moves by 1.25mm. When you switch directions, backlash starts playing a role. Also currently we stop abruptly at the end of the move so the carriage might over-travel by inertia sometimes, hope to improve that one day. Overall I didn't notice anything in the video that could cause the uneven threads you've shown 😕 Let's keep digging, I'll think what could have caused it.

Can you please post the modified h2.ino source code you've uploaded to your Arduino to https://gist.github.com/ and share the link here? Thank you

**Reply by kachurovskiy** [reply]
Re-watched the videos again, don't see anything bad, a couple times it's off by 0.08mm but I lean to explain it by the carriage over-travel due to inertia.

**Comment by Jjjigu** [comment]
Ok, i'll post this night. The version is the last week. I'll test oldest versión that i have saved.
Thanks

**Comment by Jjjigu** [comment]
Hi this is the link, only is modified steps, encoder, and velocity stepper.
I have tested oldest versions, but the problem is the same.
I am printing new gear for encoder for set to 600 (maybe the problem to set 300????) 
https://gist.github.com/Jjjigu/130d69b34927c8b10f80f709ff11ed28

**Reply by kachurovskiy** [reply]
Let's try the following test:

1. Disengage the half-nut so that cross-slide isn't moving
2. Set pitch to e.g. 1mm
3. Run spindle at constant RPM e.g. 100
4. ON the ELS

Would the stepper sound be consistent or not? If you have an RPM meter would be cool to measure the lead screw RPM. Thank you!

**Comment by Jjjigu** [comment]
Hello, this afternoon i have been disassembling the lathe because I think it may be a mechanical problem(may be on the half-nut) because it is very very strange.
 the test that you tell me I had done previously and the stepper sounds consistent.
Tonight I will be looking for some  mechanical failure tomorrow I will tell you something.
Thank you very much for your help.

**Comment by Jjjigu** [comment]
Hello after a few hours of searching I finally found the problem in the leadScrew.  This morning I tried to make a thread and it seems to come out well I have not given me time to do an intensive test but  this afternoon is that I can not do a test and post photos.
I'm testing at 800 steps because I have seen that the stepper noise is much lower, although I also understand that the speed at which you can make thread will be lower because the Arduino will not be able to send as many pulses?
 Thank you very much for your help.

**Reply by sikanderc21** [reply]
Dear sir i want to buy these control for my lathe machine can you provide me i am muhammad mati ul haq from Saudi Arabia my whatsapp is +966592391076

**Reply by Jjjigu** [reply]
> Dear sir i want to buy these control for my lathe machine can you provide me i am muhammad mati ul haq from Saudi Arabia my whatsapp is +966592391076
Hi, you must write to Maxim, is him proyect, i dont know if he sell controls

**Reply by Jjjigu** [reply]
Hi, you must write to Maxim, is his proyect, i dont know if he sell controls

**Comment by Jjjigu** [comment]
Hi, attach pictures of threads  perfects.
Thank you very much.
![IMG_20230202_065554_1](https://user-images.githubusercontent.com/121951043/216264778-529333f1-7f92-4512-9697-8e8ad4e40618.jpg)
![IMG_20230202_070738](https://user-images.githubusercontent.com/121951043/216264803-277eb138-f584-4def-8a6f-a50a65f696df.jpg)
![IMG_20230202_070717_1](https://user-images.githubusercontent.com/121951043/216264829-157b8387-0cb3-486a-8ed3-44de43783e38.jpg)

**Reply by kachurovskiy** [reply]
Awesome, congratulations 👏

**Reply by sikanderc21** [reply]
Hi sir i have make this control by self its very beautiful work everything fine and also one more i make which was before you make with Arduino mega 2560 for two axis but i little change i use lcd with I2C 2004 its working well and also change with pcb i make it on breadboard i very thankful to you and your tutorials sir please i don't bother you you make ban me from telegram before can you please release my ban from telegram thanks a lot and only one more question can it possibly if we are make any grbl code for machining and this control aslo work with grbl or mach3 etc..

**Reply by kachurovskiy** [reply]
@sikanderc21 dear friend, sorry for blocking you on Telegram, you should be unblocked now.

GRBL CNC lathe is a whole different story and I don't have plans to support it yet.

A small request to you: please kindly start a new discussion (https://github.com/kachurovskiy/nanoels/discussions/new/choose) whenever you have a new topic to talk about instead of commenting on existing threads.

All the best 😉
